src/features/user/Account/History.tsx

```diff
@@ -60,6 +78,46 @@ export default function History({
     }
   }, [paymentHistory]);
 
+  const handleIssueReceipts = async () => {
+    setLoading(true);
+    try {
+      const res = await postAuthenticatedRequest(
+        '/app/taxReceipts',
+        {
+          year: new Date().getFullYear(),
+        },
+        token,
+        handleError
+      );
+      setLoading(false);
+      if (res && res.length === 0) {
+        setError({
+          type: 'error',
+          message: t('me:taxReceiptsAlreadyGenerated'),
+        });
+        setSelectedRecord(0);
+      } else {
+        await fetchPaymentHistory();
+        setSelectedRecord(0);
+        setOpen(true)
+      }
+    } catch (err) {
+      setLoading(false);
+      console.error(err);
+    }
+  };
+
```

Comment:
Error handling logic would need refactoring once the [Error Handling PR #1647](https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/1647) is merged.

Reply:
Yes I am aware of that, since the entire setup for error handling wasn't present in develop at this point, I have structured the code in a way that will make it easier to add error handling later on.